File: Medium/3659. Partition Array Into K-Distinct Groups/solution.cs
public class Solution {
    public bool PartitionArray(int[] nums, int k) {
        int n = nums.Length;
        if(n % k != 0)return false;

        int num_group = n / k;

        Dictionary<int, int> d = new Dictionary<int, int>();
        for(int i = 0; i < n; i++){
            if(!d.ContainsKey(nums[i])){
                d[nums[i]] = 1;
            } else {
                d[nums[i]] += 1;
            }
        }

        foreach(int v in d.Values){
            if(v > num_group) return false;
        }
        return true;

    }
}

File: Medium/3659. Partition Array Into K-Distinct Groups/solution.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        if n % k != 0:
            return False
        num_group = n // k
        d = dict()
        for i in range(n):
            if nums[i] not in d:
                d[nums[i]] = 1
            else:
                d[nums[i]] += 1

        for v in d.values():
            if v > num_group:
                return False
        return True
